src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -74,12 +73,16 @@ def notifications_count
   end
 
   def fetch_notifications
-    notifications_for_subscribed_user = User.session.notifications.for_web
-    notifications = if params[:project]
-                      NotificationsFinder.new(notifications_for_subscribed_user).for_project_name(params[:project])
-                    else
-                      NotificationsFinder.new(notifications_for_subscribed_user).for_notifiable_type(params[:type])
-                    end
+    notifications_for_subscribed_user = policy_scope(Notification)
```

Comment:
To remove the duplication in the `if/else` below.

```suggestion
    notifications_for_subscribed_user = NotificationsFinder.new(policy_scope(Notification))
    if params[:project]
      notifications_for_subscribed_user.for_project_name(params[:project])
    else
      notifications_for_subscribed_user.for_notifiable_type(params[:type])
    end
```